### PR TITLE
Added AskSetUpstreamOnPush property

### PR DESCRIPTION
**Pull request by NikKovIos**
If true when push there is an al…ert to ask set upstream. If false there are no alerts, upstream setted automaticly.
Снимок экрана 2021-12-01 в 00 12 37
image

### CODE REVIEW

**Review by lucasderraugh — CHANGES_REQUESTED** [review]
Thanks for the contribution! A few things I'd change before approval:

- Could we move this setting to be just below "Commits:" section in the preferences. I think it's more logical there than after text changes.
- Let's make the text: "Ask to set upstream on push"

Other than that I'll test it out to see how it works.

**Comment by NikKovIos** [comment]
@lucasderraugh before i fix that, could you please test this functional?

File: GitUpKit/Views/GIMapViewController+Operations.m [hunk 1]
```diff
@@ -803,8 +804,10 @@ - (void)_pushLocalBranch:(GCHistoryLocalBranch*)branch toRemote:(GCRemote*)remot
             NSError* localError;
             GCHistoryLocalBranch* updatedBranch = [self.repository.history historyLocalBranchForLocalBranch:branch];  // Reload branch to check upstream!
             GCRemoteBranch* remoteBranch = [self.repository findRemoteBranchWithName:[NSString stringWithFormat:@"%@/%@", remote.name, branch.name] error:&localError];
+            BOOL askSetUpstreamOnPushSetting = [[NSUserDefaults standardUserDefaults] boolForKey:kUserDefaultsKey_AskSetUpstreamOnPush];
```
Comment (lucasderraugh) [inline GitUpKit/Views/GIMapViewController+Operations.m:807]: Let's also call this simply `askSetUpstreamOnPush` as that is more of a boolean statement.

**Review by lucasderraugh — CHANGES_REQUESTED** [review]
Looks like the radio buttons around "Diff Generation:" also go messed up. If you compare the preferences from your build to master it is offset.

Other than these issues, I think the flow itself works well though.

**Comment by NikKovIos** [comment]
- "Ask to set upstream on push" - renamed
- below "Commits" - done
- radio buttons around "Diff Generation:" - i've aligned them, there vas a bug
- askSetUpstreamOnPushSetting renamed

**Comment by NikKovIos** [comment]
Now it looks:
Снимок экрана 2021-12-01 в 15 00 38

**Comment by lucasderraugh** [comment]
The "Diff Generation:" text seems incorrectly offset with the radio buttons compared to the current layout:
Screen Shot 2021-12-01 at 8 47 13 AM
.

**Comment by NikKovIos** [comment]
@lucasderraugh indeed! It's in release now. I've fixed it in this MR and placed a screenshot. Did you try to build it?

**Comment by NikKovIos** [comment]
Did correct "Diff generation" label.

**Comment by NikKovIos** [comment]
@lucasderraugh maybe merge?)

File: GitUp/Application/Base.lproj/PreferencesWindowController.xib [hunk 2]
```diff
@@ -307,30 +307,55 @@ You must close and reopen any opened repositories in GitUp after changing this s
...
+                                            <rect key="frame" x="94" y="204" width="38" height="16"/>
+                                            <textFieldCell key="cell" lineBreakMode="clipping" title="Push:" id="adw-OE-okd">
+                                                <font key="font" usesAppearanceFont="YES"/>
+                                                <color key="textColor" name="labelColor" catalog="System" colorSpace="catalog"/>
+                                                <color key="backgroundColor" name="textBackgroundColor" catalog="System" colorSpace="catalog"/>
+                                            </textFieldCell>
+                                        </textField>
+                                        <button translatesAutoresizingMaskIntoConstraints="NO" id="hiI-QD-U0Q">
+                                            <rect key="frame" x="137" y="203" width="203" height="18"/>
+                                            <buttonCell key="cell" type="check" title="Ask to Set Upstream on push" bezelStyle="regularSquare" imagePosition="left" state="on" inset="2" id="Wi7-L1-z3w">
```
Comment (lucasderraugh) [inline GitUp/Application/Base.lproj/PreferencesWindowController.xib:327]:
```suggestion
                                            <buttonCell key="cell" type="check" title="Ask to set upstream on push" bezelStyle="regularSquare" imagePosition="left" state="on" inset="2" id="Wi7-L1-z3w">
```

**Review by lucasderraugh — CHANGES_REQUESTED** [review]
Last change is to make these items lowercase, then I'll merge it in.

**Pull request merged**